sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2ScanExecBase.scala

```diff
@@ -32,8 +32,13 @@ import org.apache.spark.util.Utils
 
 trait DataSourceV2ScanExecBase extends LeafExecNode {
 
-  override lazy val metrics = Map(
-    "numOutputRows" -> SQLMetrics.createMetric(sparkContext, "number of output rows"))
+  override lazy val metrics = {
```

Comment:
Any plans to similarly support metrics in writes?

Reply:
I looked at a few V2 write nodes, but seems we don't have any SQL metrics there. I guess we don't provide metrics for writes generally now?

If there is interest to see metrics in writes, I think it is okay to work on it later.

Reply:
Looks like updating `context.taskMetrics().outputMetrics` is just in our branch. That just uses the Hadoop FS metrics collection that we use elsewhere, so it isn't metrics from the source as we want to support in this PR.

I think it would be good to follow up and support metrics on the output side. It doesn't need to be done here, but metrics are really useful.

Reply:
Sounds good to me. I can work on it  in follow up PRs.